src/pt/mangalivre/src/eu/kanade/tachiyomi/extension/pt/mangalivre/MangaLivre.kt

```diff
@@ -1,17 +1,118 @@
 import java.text.SimpleDateFormat
 import java.util.Locale
 
-class MangaLivre : Madara(
-    "Manga Livre",
-    "https://mangalivre.tv",
-    "pt-BR",
-    SimpleDateFormat("MMMM dd, yyyy", Locale("pt")),
-) {
+class MangaLivre :
+    Madara(
+        "Manga Livre",
+        "https://mangalivre.tv",
+        "pt-BR",
+        SimpleDateFormat("dd.MM.yyyy", Locale("pt")),
+    ),
+    ConfigurableSource {
 
-    override val chapterUrlSuffix: String = ""
-    override val useNewChapterEndpoint = true
     override val id: Long = 2834885536325274328
+    override val useLoadMoreRequest = LoadMoreStrategy.Always
+    override val baseUrl by lazy { getPrefBaseUrl() }
+
+    private val preferences: SharedPreferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
+
+    override val client = super.client.newBuilder()
+        .rateLimit(2)
+        .build()
+
+    override fun headersBuilder() = super.headersBuilder()
+        .set("Origin", baseUrl)
+        .set("X-Requested-With", "XMLHttpRequest")
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val html = response.body.string()
+        var document = Jsoup.parse(html)
+
+        if (document.selectFirst("li.wp-manga-chapter") == null) {
+            val mangaUrl = response.request.url.toString().trimEnd('/')
+            val headers = headersBuilder().set("Referer", mangaUrl).build()
+            val request = POST("$mangaUrl/ajax/chapters/", headers, FormBody.Builder().build())
+
+            client.newCall(request).execute().use { xhr ->
+                if (!xhr.isSuccessful) {
+                    throw IOException("Failed to fetch chapters via AJAX: ${xhr.code}. Open in WebView.")
+                }
+                document = Jsoup.parse(xhr.body.string())
+            }
+        }
+
+        val elements = document.select("li.wp-manga-chapter").ifEmpty {
+            document.select("li.chapter-li")
+        }
+
+        if (elements.isEmpty()) {
+            throw IOException("No chapters found. Open in WebView to solve Cloudflare/Turnstile.")
+        }
+
+        return elements.map { chapterFromElement(it) }
+    }
+
+    override fun pageListParse(document: Document): List<Page> {
+        val script = document.selectFirst("script:containsData(var imgs)")?.data()
+            ?: throw IOException("Images not found. Open in WebView to solve Cloudflare/Turnstile.")
+
+        val jsonString = IMAGES_REGEX.find(script)?.groupValues?.get(1)
+            ?: throw IOException("Image list not found in script.")
+
+        val jsonArray = JSONArray(jsonString)
+
+        return buildList(jsonArray.length()) {
+            for (i in 0 until jsonArray.length()) {
+                val url = jsonArray.getString(i).trim()
+                if (url.startsWith("http")) {
+                    add(Page(size, "", url))
+                }
+            }
+        }.ifEmpty { throw IOException("Image list is empty.") }
```

Comment:
Not needed. The app will notify the user if there are no pages.
```suggestion
        }
```